src/devices/machine/k056230.cpp

```diff
@@ -119,6 +152,316 @@ void k056230_device::ram_w(offs_t offset, u32 data, u32 mem_mask)
 	COMBINE_DATA(&lanc_ram[offset & 0x7ff]);
 }
 
+
+void k056230_device::set_mode(u8 data)
+{
+	switch (data & 0xf0)
+	{
+		case 0x00:
+		case 0x20:
+			m_linkid = data & 0x0f;
+			break;
+
+		default:
+			logerror("k056230-set_mode: %02x\n", data);
+			break;
+	}
+}
+
+void k056230_device::set_ctrl(u8 data)
+{
+	m_linkenable = data && 0x10;
+
+	switch (data)
+	{
+		case 0x10:
+		case 0x18:
+			// prepare for tx?
+			break;
+
+		case 0x14:
+		case 0x1c:
+			// prepare for rx? - clear bit5
+			m_status = 0x00;
+			break;
+
+		case 0x36:
+		case 0x3e:
+			// plygonet, polynetw, gticlub = 36, midnrun 3e
+			// tx mode?
+			m_txmode = 0x01;
+			comm_tick();
+			break;
+
+		case 0x37:
+		case 0x3f:
+			// rx mode? - set bit 5
+			// plygonet, polynetw, gticlub = 37, midnrun 3f
+			m_status = 0x20;
+			m_txmode = 0x00;
+			comm_tick();
+			break;
+
+		default:
+			logerror("k056230-set_ctrl: %02x\n", data);
+			break;
+	}
+}
+void k056230_device::check_sockets()
+{
+	// if async operation in progress, poll context
+	if ((m_rx_state == 1) || (m_tx_state == 1))
+		m_ioctx.poll();
+
+	// start acceptor if needed
+	if (m_localaddr && m_rx_state == 0)
+	{
+		std::error_code err;
+		m_acceptor.open(m_localaddr->protocol(), err);
+		m_acceptor.set_option(asio::ip::tcp::acceptor::reuse_address(true));
+		if (!err)
+		{
+			m_acceptor.bind(*m_localaddr, err);
+			if (!err)
+			{
+				m_acceptor.listen(1, err);
+				if (!err)
+				{
+					osd_printf_verbose("k056230: RX listen on %s\n", *m_localaddr);
+					m_acceptor.async_accept(
+							[this] (std::error_code const &err, asio::ip::tcp::socket sock)
+							{
+								if (err)
+								{
+									LOG("k056230: RX error accepting - %d %s\n", err.value(), err.message());
+									std::error_code e;
+									m_acceptor.close(e);
+									m_rx_state = 0;
+								}
+								else
+								{
+									LOG("k056230: RX connection from %s\n", sock.remote_endpoint());
+									std::error_code e;
+									m_acceptor.close(e);
+									m_sock_rx = std::move(sock);
+									m_sock_rx.non_blocking(true);
+									m_sock_rx.set_option(asio::socket_base::receive_buffer_size(524288));
+									m_sock_rx.set_option(asio::socket_base::keep_alive(true));
+									m_rx_state = 2;
```

Comment:
You don’t want to set the non-blocking option.  You want to use the async read function so you can get notifications on your I/O service when data is available using `async_read_some`.